modules/platforms/cpp/ignite/network/detail/linux/tcp_socket_client.h

```diff
@@ -0,0 +1,247 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "ignite/network/detail/linux/sockets.h"
+#include "ignite/network/detail/utils.h"
+#include "ignite/network/socket_client.h"
+
+#include <cstdint>
+#include <sstream>
+
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <netinet/tcp.h>
+#include <netdb.h>
+#include <unistd.h>
+
+
+namespace ignite::network
+{
+
+/**
+ * Socket client implementation.
+ */
+    class tcp_socket_client : public socket_client
+    {
+    public:
+        // Delete
+        tcp_socket_client(tcp_socket_client &&) = delete;
+        tcp_socket_client(const tcp_socket_client &) = delete;
+        tcp_socket_client &operator=(tcp_socket_client &&) = delete;
+        tcp_socket_client &operator=(const tcp_socket_client &) = delete;
+
+        /** Buffers size */
+        enum { BUFFER_SIZE = 0x10000 };
+
+        // Default
+        tcp_socket_client() = default;
+
+        /**
+         * Destructor.
+         */
+        ~tcp_socket_client() override {
+            internal_close();
+        }
+
+        /**
+         * Establish connection with remote TCP service.
+         *
+         * @param hostname Remote host name.
+         * @param port TCP service port.
+         * @param timeout Timeout.
+         * @return True on success.
+         */
+        bool connect(const char* hostname, std::uint16_t port, std::int32_t timeout) override
+        {
+            internal_close();
+
+            addrinfo hints{};
+
+            hints.ai_family = AF_UNSPEC;
+            hints.ai_socktype = SOCK_STREAM;
+            hints.ai_protocol = IPPROTO_TCP;
+
+            std::stringstream converter;
+            converter << port;
+            std::string strPort = converter.str();
+
+            // Resolve the server address and port
+            addrinfo *result{NULL}; // NOLINT(modernize-use-nullptr)
+            int res = getaddrinfo(hostname, strPort.c_str(), &hints, &result);
+
+            if (res != 0)
```

Comment:
Is the `res` useful, should we include it in the error message?

Reply:
Good catch. Fixed.